spec/rspec/rails/matchers/be_a_new_spec.rb

```diff
@@ -178,7 +200,7 @@ def new_record?; false; end
             expect(record).to be_a_new(String).with(foo: 'bar')
           }.to raise_error(
             "expected #{record.inspect} to be a new String and " +
-            %(attribute {"foo"=>"bar"} was not set on #{record.inspect})
+            %(attribute #{{"foo"=>"bar"}} was not set on #{record.inspect})
```

Comment:
I would extract this to a variable which is defined based on a condition and pass that to `raise_error`

Reply:
Could you please elaborate a bit more about the reasons? As I see it, the main advantage of this approach is that the string is still very much reminds the error output. OTOH, if this is extracted into variable, it might have the benefit that Rubocop is happier. Besides that, I'd need to come up with some reasonable variable name, which is always the worst task 🙈

And also is this remark specific to this hunk or in general?

Reply:
Its in general for these interpolated hunks after your rubocop comment e.g.

```
message =
  if RUBY_VERSION >= '3.4'
    %(attribute {foo: "bar"} was not set on #{record.inspect})
  else
    %(attribute {"foo"=>"bar"} was not set on #{record.inspect})
  end

...

}.to raise_error(message)
```